### ISSUE TITLE
salt.states.mount persistence error using spaces in route

### ISSUE THREAD

**Issue by martinadolfi** [saltstack/salt#27160]
When you use the state `mount.mounted` using `persist: True` and you have spaces in the name of the device, it saves the name with spaces in `/etc/fstab` but this file won't understand spaces.
For example:

```
/mnt/documents:
  mount.mounted:
    - device: //server/DATA/Factura\ Electronica/Factura\ Nacion
    - fstype: cifs
    - mkmnt: True
    - opts: credentials=/etc/credentials/srvprocess_user,iocharset=utf8,sec=ntlm
    - dump: 0
    - pass_num: 0
    - persist: True
    - mkmnt: True
    - require:
      - file: /etc/credentials/srvprocess_user
      - pkg: cifs-utils
      - file: /mnt/documents-dir
```

This example will mount the desired network share, but also add to fstab this line:

```
//server/DATA/Factura\ Electronica/Factura\ Nacion          /mnt/documents  cifs    credentials=/etc/credentials/srvprocess_user,iocharset=utf8,sec=ntlm 0 0
```

This will result in not mounting the share again when the system is restarted, because fstab needs another interpretation for whitespaces, is should be like this>

```
//srvgoogle3/DATA/Factura\040Electronica/Factura\040Nacion          /mnt/documents  cifs    credentials=/etc/credentials/srvprocess_user,iocharset=utf8,sec=ntlm 0 0
```

Maybe the best fix is to replace `'\ '` with `'\040'` when writing this strings to fstab.

**Comment by jfindlay** [comment]
@martinadolfi, thanks for the report.  You are welcome to submit a patch with your proposed change to the 2015.5 branch.

**Comment by rallytime** [comment]
Fixed with #39593 and #45232.
